src/main/java/appeng/core/AELog.java

```diff
@@ -19,93 +19,374 @@
 package appeng.core;
 
 
-import org.apache.logging.log4j.Level;
+import javax.annotation.Nonnull;
 
-import cpw.mods.fml.relauncher.FMLRelaunchLog;
+import org.apache.logging.log4j.Level;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.logging.log4j.message.ParameterizedMessage;
 
 import appeng.core.features.AEFeature;
 import appeng.tile.AEBaseTile;
 import appeng.util.Platform;
 
 
+/**
+ * Utility class for easier logging.
+ */
 public final class AELog
 {
 
-	public static final FMLRelaunchLog INSTANCE = FMLRelaunchLog.log;
+	private static final String LOGGER_PREFIX = "AE2:";
+	private static final String SERVER_SUFFIX = "S";
+	private static final String CLIENT_SUFFIX = "C";
+
+	private static final String BLOCK_UPDATE = "Block Update of %s @ ( %d, %d, %d )";
+
+	private static final String DEFAULT_EXCEPTION_MESSAGE = "Exception: ";
 
 	private AELog()
 	{
 	}
 
-	public static void warning( final String format, final Object... data )
+	/**
+	 * Returns a {@link Logger} logger suitable for the effective side (client/server).
+	 *
+	 * @return a suitable logger instance
+	 */
+	private static Logger getLogger()
+	{
+		final String loggerName = LOGGER_PREFIX + ( Platform.isServer() ? SERVER_SUFFIX : CLIENT_SUFFIX );
+		final Logger logger = LogManager.getLogger( loggerName );
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Immediately return this expression instead of assigning it to the temporary variable "logger". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.orod.org/coding_rules#rule_key=squid%3AS1488)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Make the "logger" logger private static final and rename it to comply with the format "LOG(?:GER)?". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.orod.org/coding_rules#rule_key=squid%3AS1312)